salt/states/file.py

```diff
@@ -352,6 +352,33 @@ def _check_user(user, group):
     return err
 
 
+def _is_valid_relpath(
+        relpath,
+        maxdepth=None):
+    '''
+    Performs basic sanity checks on a relative path.
+
+    Requires slash-separated path components (i.e. the kind obtained
+    through cp.list_master or other such calls).
+
+    Ensures that the path does not contain directory transversal, and
+    that it does not exceed a stated maximum depth (if specified).
+    '''
+    # Check relpath surrounded by slashes, so that `..` can be caught as
+    # a path component at the start, end, and in the middle of the path.
+    if '/../' in '/{0}/'.format(relpath):
```

Comment:
This will not work in Windows, since ``os.path.relpath()`` uses a backslash as a path separator on that platform. ``/`` should be replaced with ``os.sep``.

Also, can you explain the reasoning behind putting a slash after the ``..``? Your comment above is not very clear. ``os.path.relpath()`` doesn't include these for directories. However, ``file.recurse`` should just operate on file paths, and create directories as needed. Maybe that's the reason?